ls1intum/Artemis
# [Bugfix] Fix parsing of test-case names in markdown

### Checklist

### Motivation and Context



There were several parsing errors with test case names including "()" and "," characters like "testBubbleSort(IOTester,IOTester)"

### Description


Improve the regular expressions used for parsing the test case names.

### Steps for Testing


1. Log in to Artemis
2. Create a Progamming Exercises and update the names of the Test Cases to include parameters like "testname(param1, param2)"
3. Go to the Instructor Code Editor and edit the exercise instructions accordingly.
4. View the rendered markdown of the exercise instructions. The tasks and uml diagram should be parsed and rendered correctly.

### Test Coverage


Nothing changed.

### Screenshots



**Markdown for tasks and uml**
<img width="466" alt="Bildschirmfoto 2020-11-02 um 22 11 06" src="https://user-images.githubusercontent.com/13920539/97919634-68f2c400-1d58-11eb-8e9c-2524125c88a5.png">
<img width="493" alt="Bildschirmfoto 2020-11-02 um 22 11 21" src="https://user-images.githubusercontent.com/13920539/97919640-6a23f100-1d58-11eb-84a6-e1aff3e1f4bf.png">
